### PR TITLE
Issue 1 & 3

### PR DESCRIPTION
**Pull request by JaySunSyn**
Wrong path displayed for search results #3  got solved. The wrong path was mainly displayed when searching a node from the root level. 

List goes below viewport height when filled #1 
Also, issue 1 got solved while working on Performance. Thats why, no additional PR for this one. Hope thats fine.

### CODE REVIEW

File: cosmoz-treenode-navigator.js [hunk 1]
```diff
@@ -19,8 +19,9 @@
 			},
 			dataPlane: {
 				type: Array,
-				computed: '_computeDataPlane(_searching, inputValue, _renderedLevel, _openNodeLevelPathParts, data)',
-				notify: true
+				value: function() { return []; },
```
Comment (nomego) [inline cosmoz-treenode-navigator.js]: This shouldn't be needed if it is computed.

Comment (JaySunSyn) [inline cosmoz-treenode-navigator.js]: Yes, true. This is still from a version of mine with observing and then setting dataPlane but returned to computed one.

File: cosmoz-treenode-navigator.js [hunk 2]
```diff
@@ -141,40 +142,120 @@
...
 			if (searching) {
 				return this.searchAllBranches(inputValue, openNodeLevelPathParts, renderedLevel, data);
 			}
 			return renderedLevel;
 		},
+
+		/**
+		 * Gets called if a user selects a node.
+		 */
+		_nodeSelected: function(e) {
```
Comment (nomego) [inline cosmoz-treenode-navigator.js]: space

File: cosmoz-treenode-navigator.js [hunk 3]
```diff
@@ -141,40 +142,120 @@
...
+			this.highlightedNodePath = e.currentTarget.locationPath;
```
Comment (nomego) [inline cosmoz-treenode-navigator.js]: Any reason why doing this instead of `event.model` ?

File: cosmoz-treenode-navigator.js [hunk 4]
```diff
@@ -141,40 +142,120 @@
...
-				children = nodes,
-				childPath,
-				level;
+			var pathSegment = nodes,
+				node,
+				level = [];
 
-			if (node) {
-				children = node[this.childProperty];
+			if (pl==='' || typeof pl=== 'undefined') {
```
Comment (nomego) [inline cosmoz-treenode-navigator.js]: spacing

**Comment by JaySunSyn** [comment]
@nomego please let me know if the "space problem" got resolved or elaborate. Thanks

**Comment by JaySunSyn** [comment]
@nomego Please check, thanks. I understand when you write spacing missing, you mean `function (x)` instead of `function(x)` or do I get this wrongly?

File: cosmoz-treenode-navigator.js [hunk 5]
```diff
@@ -196,70 +268,65 @@
...
-			// keep our own copy
-			var pathParts = pathPartsRaw.slice(),
-				localSection = '',
-				results = [];
+			var results = [];
+			this._findInNodes(nodeList, searchWord, this.comparisonProperty, results, this.childProperty, data);			
+			return results;
+		},
+		_getPathString: function (pl, nodeList) {
+			var nodesOnPath = this._getNodesOnPath(pl, nodeList);
```
Comment (nomego) [inline cosmoz-treenode-navigator.js]: One var statement

File: cosmoz-treenode-navigator.js [hunk 6]
```diff
@@ -196,70 +268,65 @@
...
+		 */
+		_findInNodes: function (nodes, searchStr, searchAttr, results, childProperty, data) {
+		    var arr, value, children, pl;
 
-			if (pathParts.length > 0) {
-				localSection = pathParts.map(function (part) {
-					return part.name;
-				}).join(' / ');
-			}
+		    if (nodes instanceof Array) {
```
Comment (nomego) [inline cosmoz-treenode-navigator.js]: Use `Array.isArray()`

File: cosmoz-treenode-navigator.js [hunk 7]
```diff
@@ -196,70 +268,65 @@
...
+				nodes.forEach(function () {
+					this._findInNodes(nodes[i], searchStr, searchAttr, results, childProperty, data);
+				}, this)
```
Comment (nomego) [inline cosmoz-treenode-navigator.js]: return here and drop the else

File: cosmoz-treenode-navigator.js [hunk 8]
```diff
@@ -196,70 +268,65 @@
...
-				if (node.children && Object.keys(node.children).length > 0) {
-					pathParts.push(node);
-					results = results.concat(this.searchAllBranches(searchWord, pathParts, this._renderLevel(node.path, data), data));
-				}
-			}, this);
-			return results;
+		    	if (nodes[searchAttr].toLowerCase().indexOf(searchStr.toLowerCase()) !== -1) {
+		    		nodes.path = nodes.pathLocator || nodes.path;
+		    		nodes.sectionName = this._getPathString(nodes.path, data);
+		            results.push(nodes);
```
Comment (nomego) [inline cosmoz-treenode-navigator.js]: mixed spaces and tabs

**Pull request merged**